### PR TITLE
Add some commentary around how dates are formatted

### PR DESCRIPTION
**Pull request by alexwlchan**
## Who is this for?

Devs.

## What is it doing for them?

Helping them see what this code does. You can see it from the tests, but IMO it's useful to have the intended behaviour described explicitly.

### CODE REVIEW

File: common/views/components/OpeningTimes/OpeningTimes.tsx [hunk 1]
```diff
@@ -5,46 +5,42 @@ import {
...
   getNameFromCollectionVenue,
 } from '@weco/common/data/hardcoded-ids';
-import { FunctionComponent, ReactElement } from 'react';
+import { FC } from 'react';
 
 type Props = {
   venues: Venue[];
 };
 
-const OpeningTimes: FunctionComponent<Props> = ({
```
Comment (alexwlchan) [inline common/views/components/OpeningTimes/OpeningTimes.tsx:14]: Although a big diff, this is just removing some unnecessary type signatures and removing the `return` from the body of this component.

File: common/views/components/Dot/Dot.tsx [hunk 2]
```diff
@@ -17,8 +17,8 @@ type Props = {
   color: PaletteColor;
 };
 
-const Dot: FunctionComponent<Props> = ({ color }: Props) => {
-  return <DotEl className={classNames({ [`font-${color}`]: true })} />;
-};
+const Dot: FC<Props> = ({ color }) => (
+  <DotEl className={classNames({ [`font-${color}`]: true })} />
+);
```
Comment (alexwlchan) [inline common/views/components/Dot/Dot.tsx:22]: this is completely irrelevant, but I wanted to trigger a new Chromatic build

File: common/utils/format-date.ts [hunk 3]
```diff
@@ -8,36 +8,49 @@ export function formatLondon(
...
 
+/** Formats a date as the day of the week, plus the day of the month and the name of the month.
+ *
+ * e.g. 'Thursday 29 September'
+ */
 export function formatDayDate(date: Date): string {
   return `${formatDay(date)} ${formatDate(date)}`;
 }
 
+/** Formats a date as the day of the month, the name of the month, and the yea.r
```
Comment (rcantin-w) [inline common/utils/format-date.ts:24]: nit, typo at `year`

File: common/views/components/Dot/Dot.tsx (hunk 2 above)
Comment (rcantin-w) [reply common/views/components/Dot/Dot.tsx:22]: I see both FC and FunctionComponent used and I've been trying to figure out why one and not the other, internet seems to lean towards FunctionComponent, do we have a preference?

**Review by rcantin-w — APPROVED** [review]
Useful one thanks!

File: common/views/components/Dot/Dot.tsx (hunk 2 above)
Comment (alexwlchan) [reply common/views/components/Dot/Dot.tsx:22]: I feel like we've been leaning towards `FC` in our codebases. I don't think it makes a huge difference; we should pick one and standardise on it.

**Pull request merged**